galaxy_ng/app/api/ui/views/auth.py

```diff
@@ -0,0 +1,69 @@
+from django.contrib import auth as django_auth
+from django.db import transaction
+from django.http import Http404
+
+from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+from rest_framework import status as http_code
+
+from galaxy_ng.app.api import base as api_base
+from galaxy_ng.app.api.permissions import RestrictOnCloudDeployments
+from galaxy_ng.app.api.ui.serializers import LoginSerializer
+
+
+__all__ = (
+    'LoginView',
+    'LogoutView',
+    'TokenView',
+)
+
+
+class LoginView(api_base.GenericAPIView):
+    serializer_class = LoginSerializer
+    permission_classes = (RestrictOnCloudDeployments,)
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+
+        username = serializer.validated_data['username']
+        password = serializer.validated_data['password']
```

Comment:
This isn't a requirement, but it would be good to have a `remember me` so that users can choose to not remain logged in.